File: C#/Sum Of Multiples/SumOfMultiples.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class SumOfMultiples
{
    public static List<int> Multiples(int n, int limit)
    {
        var output = new List<int>();

        for (int i = n; i < limit; i++)
        {
            if (i % n == 0)
            {
                output.Add(i);
            }
        }

        return output;
    }

    public static int Sum(IEnumerable<int> multiples, int max)
    {
        var hs = new HashSet<int>();

        foreach (var item in multiples)
        {
            if (item != 0)
            {
                List<int> numbers = Multiples(item, max);
                foreach (var number in numbers)
                {
                    hs.Add(number);
                }
            }
        }

        return hs.Sum();
    }
}

File: TypeScript/sum-of-multiples.ts
function multiples(n: number, limit: number): number[] {
    let output = [];

    for (let i = n; i < limit; i++) {
        if(i % n == 0) {
            output.push(i);
        }
    }

    return output;
}

export const sum = (items: number[], levelNum: number): number => {
    let s = new Set<number>();
    
    items.forEach(item => {
        let nums = multiples(item, levelNum);
        nums.forEach(n => s.add(n)); 
    });

    return Array.from(s).reduce((a, b) => a + b, 0);
};
  
console.log(sum([3, 5], 20));      // 78
console.log(sum([43, 47], 10000)); // 2203160
console.log(sum([3, 5], 1));       // 0